src/contexts/states/viewStore.ts

```diff
@@ -0,0 +1,246 @@
+            // Clamp is to prevent someone from requesting non-existent page.
+            newPageNum = clamp(navAction.args.pageNum, 1, numPages);
+            break;
+        case ACTION_NAME.FIRST_PAGE:
+            position = EVENT_POSITION_ON_PAGE.TOP;
+            newPageNum = 1;
+            break;
+        case ACTION_NAME.PREV_PAGE:
+            position = EVENT_POSITION_ON_PAGE.BOTTOM;
+            newPageNum = clamp(currentPageNum - 1, 1, numPages);
+            break;
+        case ACTION_NAME.NEXT_PAGE:
+            position = EVENT_POSITION_ON_PAGE.TOP;
+            newPageNum = clamp(currentPageNum + 1, 1, numPages);
+            break;
+        case ACTION_NAME.LAST_PAGE:
+            position = EVENT_POSITION_ON_PAGE.BOTTOM;
+            newPageNum = numPages;
+            break;
+        default:
+            return null;
+    }
+
+    return {
+        code: CURSOR_CODE.PAGE_NUM,
+        args: {pageNum: newPageNum, eventPositionOnPage: position},
+    };
+};
+
+// eslint-disable-next-line max-lines-per-function
+const useViewStore = create<ViewState>((set, get) => ({
+    ...VIEW_STORE_DEFAULT,
+    filterLogs: (filter: LogLevelFilter) => {
+        const {setUiState} = useUiStore.getState();
+        setUiState(UI_STATE.FAST_LOADING);
+        const {logEventNum} = useContextStore.getState();
+
+        useLogFileManagerStore
+            .getState()
+            .logFileManagerProxy.setFilter(
+                {code: CURSOR_CODE.EVENT_NUM,
+                    args: {
+                        eventNum: 0 === logEventNum ?
+                            1 :
+                            logEventNum,
+                    }},
+                get().isPrettified,
+                filter
+            ).then((pageData: PageData) => {
+                useViewStore.getState().updatePageData(pageData);
+            })
+            .catch((e: unknown) => {
+                useContextStore.getState().postPopUp({
+                    level: LOG_LEVEL.ERROR,
+                    message: String(e),
+                    timeoutMillis: DO_NOT_TIMEOUT_VALUE,
+                    title: "Action failed",
+                });
+            });
+        useQueryStore.getState().startQuery();
+    },
+    loadPageByAction: (navAction: NavigationAction) => {
+        const {logEventNum} = useContextStore.getState();
+        const {fileSrc, loadFile} = useLogFileStore.getState();
+        if (navAction.code === ACTION_NAME.RELOAD) {
+            if (null === fileSrc || 0 === logEventNum) {
+                throw new Error(
+                    `Unexpected fileSrc=${JSON.stringify(
+                        fileSrc
+                    )}, logEventNum=${logEventNum} when reloading.`
+                );
+            }
+            loadFile(fileSrc, {
+                code: CURSOR_CODE.EVENT_NUM,
+                args: {eventNum: logEventNum},
+            });
+
+            return;
+        }
+
+        const {isPrettified, numPages, pageNum} = get();
```

Comment:
ditto: state import from the same store should be the very first state import line at the function body top